File: STRINGS/1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.swift
class Solution {
    func maxDepth(_ s: String) -> Int {
        var st = [Int]()
        var maxCounter = 0
        
        for char in s {
            if char == "(" {
                st.append(1)
            } else if char == ")" {
                st.popLast()
            }
            maxCounter = max(maxCounter, st.count)
        }
        
        return maxCounter
    }
}

File: STRINGS/1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.php
class Solution {
    /**
     * @param String $s
     * @return Integer
     */
    function maxDepth($s) {
        $st = [];
        $maxCounter = 0;

        for ($i = 0; $i < strlen($s); $i++) {
            if ($s[$i] === '(') {
                array_push($st, $i);
            } else if ($s[$i] === ')') {
                array_pop($st);
            }
            $maxCounter = max($maxCounter, count($st));
        }

        return $maxCounter;
    }
}
